ckanext/ytp/request/tests/test_plugin.py

```diff
@@ -1,44 +1,65 @@
-"""Tests for plugin.py."""
 # encoding: utf-8
+import logging
+from nose.tools import assert_true, assert_equal
 
-# from nose.tools import assert_raises
-import ckan.model as model
-import ckan.plugins
-# import ckan.tests.factories as factories
-# import ckan.logic as logic
+import mock
+import __builtin__ as builtins
+from bs4 import BeautifulSoup
+from ckan.lib.helpers import url_for
+from ckan.common import config
 
+import ckan.model as model
+import ckan.tests.helpers as helpers
+import ckan.tests.factories as factories
+from ckan.model.system_info import get_system_info
+from ckan.tests.helpers import submit_and_follow
 
-class TestYTPRequestsPlugin(object):
-    '''Tests for the ckanext.example_iauthfunctions.plugin module.
+webtest_submit = helpers.webtest_submit
+PLUGIN_CONTROLLER = 'ckanext.ytp.request.controller:YtpRequestController'
 
-    Specifically tests that overriding parent auth functions will cause
-    child auth functions to use the overridden version.
+class TestViewingActionedReferral(helpers.FunctionalTestBase):
+    '''
+        Test that viewing an already actioned membership request does not 404
     '''
-    @classmethod
-    def setup_class(cls):
-        '''Nose runs this method once to setup our test class.'''
-        # Test code should use CKAN's plugins.load() function to load plugins
-        # to be tested.
-        ckan.plugins.load('ytp_request')
-
-    def teardown(self):
-        '''Nose runs this method after each test method in our test class.'''
-        # Rebuild CKAN's database after each test method, so that each test
-        # method runs with a clean slate.
-        model.repo.rebuild_db()
-
-    @classmethod
-    def teardown_class(cls):
-        '''
-        Nose runs this method once after all the test methods in our class
-        have been run.
-        '''
-        # We have to unload the plugin we loaded, so it doesn't affect any
-        # tests that run after ours.
-        ckan.plugins.unload('ytp_request')
-
-    def sample_test(self):
-        '''
-        Placeholder test.
-        '''
-        pass
+
+    _load_plugins = (u'ytp_request', )
+
+    @mock.patch('ckanext.ytp.request.logic.action.create.flash_success')
+    @mock.patch('ckanext.ytp.request.logic.action.update.flash_success')
```

Comment:
I think you could get rid of the mocks if you'd use the test app to call the actions. Example:
```
        app = helpers._get_test_app()

        with app.flask_app.test_request_context():
            app.get(
                url_for(
                    controller='group',
                    action='member_new',
                    id=group['id'],
                ),
                status=403,
            
```
But we're dealing with pylons here so I'm not sure how exactly to do that. Let's leave it right now.